### ISSUE TITLE
Diagnostics for `reportUnnecessaryTypeIgnoreComment` are not marked as such

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#7883]
Reproducible example ([playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&reportUnnecessaryTypeIgnoreComment=true&code=IYLgBAlgdgLmC8YDkSxgMRgA4E8BOEA5gBYzhFQD2eApmANoCCAugLABQQA)):

```python
a: int = ''  # pyright: ignore [A]
#                               ~ Unnecessary "# pyright: ignore" rule: "A"
```

**Comment by erictraut** [comment]
I don't understand what issue you're reporting here. Pyright does generate an error for an unnecessary type ignore comment in this case, as it is designed to do.

Here's the screen shot from the pyright playground.

![image](https://github.com/microsoft/pyright/assets/7040122/6f832884-e7fe-4f4e-b7dc-d90d9e3c0234)

**Comment by erictraut** [comment]
Ah, I see what you mean. Yep, I agree that's a bug. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
